spring-ai-alibaba-core/src/main/java/com/alibaba/cloud/ai/dashscope/rag/AnalyticdbVector.java

```diff
@@ -0,0 +1,270 @@
+
+	}
+
+	private void createNameSpaceIfNotExists() throws Exception {
+		try {
+			DescribeNamespaceRequest request = new DescribeNamespaceRequest()
+				.setDBInstanceId(this.config.getDBInstanceId())
+				.setRegionId(this.config.getRegionId())
+				.setNamespace(this.config.getNamespace())
+				.setManagerAccount(this.config.getManagerAccount())
+				.setManagerAccountPassword(this.config.getManagerAccountPassword());
+			this.client.describeNamespace(request);
+		}
+		catch (TeaException e) {
+			if (Objects.equals(e.getStatusCode(), 404)) {
+				CreateNamespaceRequest request = new CreateNamespaceRequest()
+					.setDBInstanceId(this.config.getDBInstanceId())
+					.setRegionId(this.config.getRegionId())
+					.setNamespace(this.config.getNamespace())
+					.setManagerAccount(this.config.getManagerAccount())
+					.setManagerAccountPassword(this.config.getManagerAccountPassword())
+					.setNamespacePassword(this.config.getNamespacePassword());
+				this.client.createNamespace(request);
+			}
+			else {
+				throw new Exception("failed to create namespace:{}", e);
+			}
+		}
+	}
+
+	private void createCollectionIfNotExists(Long embeddingDimension) throws Exception {
+		try {
+			// Describe the collection to check if it exists
+			DescribeCollectionRequest describeRequest = new DescribeCollectionRequest()
+				.setDBInstanceId(this.config.getDBInstanceId())
+				.setRegionId(this.config.getRegionId())
+				.setNamespace(this.config.getNamespace())
+				.setNamespacePassword(this.config.getNamespacePassword())
+				.setCollection(this.collectionName);
+			this.client.describeCollection(describeRequest);
+			logger.debug("collection" + this.collectionName + "already exists");
+		}
+		catch (TeaException e) {
+			if (Objects.equals(e.getStatusCode(), 404)) {
+				// Collection does not exist, create it
+				String metadata = JSON.toJSONString(new JSONObject().fluentPut("refDocId", "text")
+					.fluentPut("content", "text")
+					.fluentPut("metadata", "jsonb"));
+				String fullTextRetrievalFields = "content";
+				CreateCollectionRequest createRequest = new CreateCollectionRequest()
+					.setDBInstanceId(this.config.getDBInstanceId())
+					.setRegionId(this.config.getRegionId())
+					.setManagerAccount(this.config.getManagerAccount())
+					.setManagerAccountPassword(this.config.getManagerAccountPassword())
+					.setNamespace(this.config.getNamespace())
+					.setCollection(this.collectionName)
+					.setDimension(embeddingDimension)
+					.setMetrics(this.config.getMetrics())
+					.setMetadata(metadata)
+					.setFullTextRetrievalFields(fullTextRetrievalFields);
+				this.client.createCollection(createRequest);
+				logger.debug("collection" + this.collectionName + "created");
+			}
+			else {
+				throw new RuntimeException(
+						"Failed to create collection " + this.collectionName + ": " + e.getMessage());
+			}
+		}
+	}
+
+	@Override
+	public void add(List<Document> texts) {
+		try {
+			initialize();
```

Comment:
How about using `afterPropertiesSet()` to do initialization. It guarantees only execute once and before any other method calls.